text/vga-console.md

```diff
@@ -0,0 +1,106 @@
+# VGA Console
+
+The VGA spec and the guaranteed presence of VGA hardware on any machine or VM
+intermezzOS targets provide convenient methods of printing information for us to
+observe the kernel in action.
+
+## Details
+
+The VGA spec says that the magic address `0xB8000` is the start of video memory,
+and represents the top left cell on the screen. The screen is 80 columns wide,
+25 rows high, and each column is two bytes. In total, VGA video memory is 4000
+bytes (just under 4K).
+
+### Overview
+
+At present, intermezzOS' VGA implementation consists of a struct that owns a
+VGA-sized buffer **not** at the magic location, the VGA memory that is at the
+magic location, and some minor internal state.
+
+The console structure uses the buffer to collect text until `flush()` is called,
+at which point the buffer is copied en masse to the video memory. This provides
+apparently-atomic updates to the screen, although this operation need not be
+*actually* atomic as (a) good design and Rust checks should guarantee only one
+VGA structure owns video memory, and (b) moving 4k of memory is a long time to
+not have any other interrupts.
+
+### Proposed Design
+
+Philipp Oppermann suggests using a Unique pointer to wrap the magic memory
+location so that the compiler can guarantee that there is ever only one mutable
+owner of the video memory. This is a good plan, but in practice may be overkill
+for intermezzOS' needs and goals. Unique is yet another feature of Rust nightly
+that needs to be set as a crate-level attribute, and while intermezzOS already
+makes plentiful use of such features, I feel that where some can be trimmed, we
+may want to do so.
```

Comment:
Yes, I'm torn about this myself, which is why I didn't initially use Unique. I'm not opposed to it though.